great_expectations/data_context/data_context.py

```diff
@@ -788,7 +788,7 @@ def get_config_with_variables_substituted(self, config=None) -> DataContextConfi
             config = self._project_config
 
         substituted_config_variables = substitute_all_config_variables(
-            dict(self._load_config_variables_file()),
+            self.config_variables,
```

Comment:
Same comment as in checkpoint.py -- we should probably use `BaseDataContext. DOLLAR_SIGN_ESCAPE_STRING`.

Reply:
Agreed - also adjusted.